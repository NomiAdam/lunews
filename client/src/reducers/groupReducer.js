import { evolve, always } from 'ramda';
import { createReducer } from '../context/util';

export const groupActionTypes = {
    selectGroup: 'news/SELECT_GROUP',
    selectMessage: 'news/SELECT_MESSAGE',
    selectFocused: 'news/SELECT_FOCUSED',
    selectStarred: 'news/SELECT_STARRED',
};

export default DEFAULT_STATE => createReducer(DEFAULT_STATE, {
    [ groupActionTypes.selectGroup ]: (state, { payload } ) => ({
        ...state,
        group: evolve({ selected: always(payload) }, state.group),
    }),
    [ groupActionTypes.selectMessage ]: (state, { payload } ) => ({
        ...state,
        group: evolve({ messageID: always(payload) }, state.group),
    }),
    [ groupActionTypes.selectFocused ]: (state, { payload } ) => ({
        ...state,
        group: evolve({ focusedMessage: always(payload) }, state.group),
    }),
    [ groupActionTypes.selectStarred ]: (state, { payload } ) => ({
        ...state,
        group: evolve({ isStarred: always(payload) }, state.group),
    }),
});
