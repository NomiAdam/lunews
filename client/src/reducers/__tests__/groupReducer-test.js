/* eslint-disable no-undef */
import { setGroupThreadCount, setActiveGroup } from '../groupReducer';

const mockPayload = 'mock';

describe('groupReducer', () => {
    describe('groupReducerActionCreators', () => {
        it('should match the snapshot for setGroupThreadCount', () => {
            expect(setGroupThreadCount(mockPayload)).toMatchSnapshot();
        });
        it('should match the snapshot for setActiveGroup', () => {
            expect(setActiveGroup(mockPayload)).toMatchSnapshot();
        });
    });
});
