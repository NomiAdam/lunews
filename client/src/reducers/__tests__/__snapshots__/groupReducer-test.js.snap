// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`groupReducer groupReducerActionCreators should match the snapshot for setActiveGroup 1`] = `
Object {
  "payload": "mock",
  "type": "news/SELECT_GROUP",
}
`;

exports[`groupReducer groupReducerActionCreators should match the snapshot for setGroupThreadCount 1`] = `
Object {
  "payload": "mock",
  "type": "news/FETCH_THREAD_COUNT",
}
`;
